### ISSUE TITLE
Add function that returns important object properties for easier documentation (Edited)

### ISSUE THREAD

**Issue by gherold** [acoular/acoular#156]
Former title: ~~Add optional parameter for digest function that makes it return the important object properties~~

~~For easier result storing, it would be helpful if the (already existing) digest function had a feature to actually return the values of the observed traits it is fed (e.g. as dictionary).~~

For easy data processing documentation and reproducibility, it would be nice to have a function that extracts the important (i.e. digested) trait values, e.g. as dictionary.

This would make it much easier to export results with all the parameters that went into generating them.
Currently, this has to be done by hand, which is tedious.

**Comment by esarradj** [comment]
Could you describe a use case for this? Probably we would need a different function. Normally `__repr__` serves this purpose.

**Comment by gherold** [comment]
Yes, `__repr__` could be used for this. However, the digest already know what's important for the result and only one function would have to be changed instead of adapting each class.

It would benefit the reproducibility of (cached) data and make saving parameters of parameter studies very easy.

**Comment by esarradj** [comment]
If the output of digest function is changed, the whole digest mechanism has to be revisited because the digest frequently depends on the output of the digest function of another class. A new repr on the other hand needs just to be defined once: either as a mixin class, or a separate function to called with the object as an argument. Depends in the use case ...

**Comment by gherold** [comment]
How would `repr` get access to the digest's `depends_on` parameter, which is necessary? If the `_get_digest` function gets an optional parameter which only changes the output if set, no harm would be done to the existing mechanism.

If there is a possibility to access the `depends_on` parameter from the outside, this would maybe be preferable, as the solution would not be as hackish.

**Comment by gherold** [comment]
In connection with this, also #153 should be considered, as this might impact what the optimal path here would be.

**Comment by esarradj** [comment]
The list of depending trait names of any `<obj>` is easy to access using `<obj>.trait('digest').depends_on`. 

A custom repr could just iterate through this and generate some meaningful printout string. It has to be decided how to deal with digests the digest depends on. They can either be printed or instead the repr of the depending-on object can be given (in a recursive manner).
